handler/src/main/java/io/netty/handler/ssl/EnhancingX509ExtendedTrustManager.java

```diff
@@ -21,15 +21,24 @@
 import java.security.cert.X509Certificate;
 import java.util.Collection;
 import java.util.List;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.X509ExtendedTrustManager;
 import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
 
 /**
  * Wraps an existing {@link X509ExtendedTrustManager} and enhances the {@link CertificateException} that is thrown
  * because of hostname validation.
  */
 final class EnhancingX509ExtendedTrustManager extends X509ExtendedTrustManager {
+
+    // Constants for subject alt names of type DNS and IP. See X509Certificate#getSubjectAlternativeNames() javadocs.
+    static final int ALTNAME_DNS = 2;
+    static final int ALTNAME_IP = 7;
```

Comment:
Consider adding URI names to the mix (code `6`). These are common in some environments with gRPC services that use SPIFFEs. Though the hostname matcher won't be looking at them, having them there can help debugging cases where hostname verification was enabled when it shouldn't be.